packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -331,12 +334,32 @@ export function calculatePositionInternal(
 
   overlaySize.height = Math.min(overlaySize.height, maxHeight);
 
-  position = computePosition(childOffset, boundaryDimensions, overlaySize, placementInfo, normalizedOffset, crossOffset, containerOffsetWithBoundary, isContainerPositioned);
+  position = computePosition(childOffset, boundaryDimensions, overlaySize, placementInfo, normalizedOffset, crossOffset, containerOffsetWithBoundary, isContainerPositioned, arrowCrossSize);
   delta = getDelta(crossAxis, position[crossAxis], overlaySize[crossSize], boundaryDimensions, padding);
   position[crossAxis] += delta;
 
   let arrowPosition: Position = {};
-  arrowPosition[crossAxis] = (childOffset[crossAxis] - position[crossAxis] + childOffset[crossSize] / 2);
+
+  // All values are transformed so that 0 is at the top/left of the overlay depending on the orientation
+  // Prefer the arrow being in the center of the trigger/overlay anchor element
+  let preferredArrowPosition = childOffset[crossAxis] + .5 * childOffset[crossSize] - overlaySize[crossAxis];
+
+  // Min/Max position limits for the arrow with respect to the overlay
+  const arrowMinPosition = arrowCrossSize / 2;
+  const arrowMaxPosition = overlaySize[crossSize] - (arrowCrossSize / 2);
+
+  if (preferredArrowPosition < arrowMinPosition || preferredArrowPosition > arrowMaxPosition) {
+    // Prefer the arrow being in the center of the overlay if placing it in the center of the button would mean it doesn't also overlap the overlay
+    preferredArrowPosition = overlaySize[crossSize] / 2;
+  }
+
+  // Min/Max position limits for the arrow with respect to the trigger/overlay anchor element
+  const arrowOverlappingChildMinEdge = childOffset[crossAxis] - overlaySize[crossAxis] + (arrowCrossSize / 2);
+  const arrowOverlappingChildMaxEdge = childOffset[crossAxis] + childOffset[crossSize] - overlaySize[crossAxis] - (arrowCrossSize / 2);
```

Comment:
Open question for the team about what boundaries we want for the arrow tip positioning vs the trigger element's edges. At the moment the `(arrowCrossSize / 2)` makes it so the entire arrow must fit in the trigger, with some additional fudge factor applied in Popover/TooltipTrigger so it doesn't quite hit the edges of the overlay/trigger.

If we are ok with something like the following
<img width="245" alt="image (5)" src="https://user-images.githubusercontent.com/9661127/225782111-c1565778-5b9f-4dc3-b3c5-b27f89c8f1ef.png">
I can get rid of the `(arrowCrossSize / 2)` here

Reply:
I don't quite follow, maybe we can spend a little time tomorrow looking at this after standup if my suggestions don't address this?